pkg/storage/unified/apistore/prepare.go

```diff
@@ -71,9 +73,19 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 	if err != nil {
 		return nil, err
 	}
+
+	if previous.GetUID() == "" {
+		klog.Errorf("object is missing UID: %s, %s", obj.GetGroupVersionKind().String(), obj.GetName())
+	}
+
+	if obj.GetName() != previous.GetName() {
+		return nil, fmt.Errorf("name mismatch between existing and updated object")
+	}
+
 	obj.SetUID(previous.GetUID())
 	obj.SetCreatedBy(previous.GetCreatedBy())
 	obj.SetCreationTimestamp(previous.GetCreationTimestamp())
+	obj.SetResourceVersion("")
```

Comment:
This doesn't seem write. We shouldn't clear the resource version on update 🤔. 
The RV is needed for the optimistic concurrency check.

Reply:
so, from what I understand, RV, regardless if it's a create or an update, should not have an RV as this should be set my the DB: if I remove this check here, it will bouce back [here](https://github.com/grafana/grafana/pull/93852/files#diff-e4855c422ca8f979c4567e5d12f34ecd3a438a166dc8906ae3edda28bd5e4d20R281-R284)

Reply:
This is correct -- clear it from the body because it will be generated on the server and weird to include the previous value in the payload